### PR TITLE
Stac Client IO and Documentation Changes

### PR DESCRIPTION
**Pull request by GregoryPetrochenkov-NOAA**
This PR resolves the following two issues, #166 #167.

## Additions

-  Images for README
-  Files for continuous notebooks
-  tutorials.rst
-  cases_stac/test_stac.py

## Changes

- 04_EXAMPLE.MD/README.MD/PYPI_README.MD_SPHINX_README: Include continuous and catalog comparison example
- compile_readme_and_arrange_docs.py:  Add image sizes for new documents
- SPHINX_TUTORIALS: rewrite after compile
- index.rst: nest tutorials in tutorials section
- compute_continuous_metrics.py:  Streamline sentinel values for integer datasets
- loading_datasets.py:  Add stac client IO functions

### CODE REVIEW

File: README.MD [hunk 1]
```diff
@@ -69,8 +71,105 @@ benchmark = rxr.open_rasterio('benchmark_map_two_class_categorical.tif', mask_an
...
+        "masked": True
+    }
+}
+
+agreement_continuous_catalog = catalog_compare(**arguments)
+```
+
+#### Catalog Outputs
+
+`agreement_map`
```
Comment (fernando-aristizabal) [inline README.MD:161]: If I remember correctly, agreement maps are written out to file somewhere or thrown in a column in the agreement catalog? Might be worth showing where that is in the example or just refer to notebook for brevity?

File: README.MD [hunk 2]
```diff
@@ -69,8 +71,105 @@ benchmark = rxr.open_rasterio('benchmark_map_two_class_categorical.tif', mask_an
 
 ![alt text](https://github.com/NOAA-OWP/gval/raw/main/docs/images/metric_table.png)
 
-For more details on how to use this software, check out this [notebook
-tutorial](https://github.com/NOAA-OWP/gval/blob/main/notebooks/Tutorial.ipynb).
+### Continuous Example
+
+The same can be done for continuous rasters as shown below (in this case
```
Comment (fernando-aristizabal) [inline README.MD:76]: for rasters with continuous valued statistical data types

File: docs/images/benchmark_catalog.png
Comment (fernando-aristizabal) [inline docs/images/benchmark_catalog.png:1]: Should we integrate an attribute column to show how this works as well?

File: docs/images/candidate_catalog.png
Comment (fernando-aristizabal) [inline docs/images/candidate_catalog.png:1]: Same comment about attribute columns?

File: src/gval/comparison/compute_continuous_metrics.py [hunk 3]
```diff
@@ -17,7 +17,7 @@
 
 from gval import ContStats
 from gval.utils.schemas import Metrics_df, Subsample_identifiers, Sample_identifiers
-from gval.utils.loading_datasets import _check_dask_array, _convert_to_dataset
+from gval.utils.loading_datasets import _convert_to_dataset
 
 
 def _get_selected_datasets(
```
Comment (fernando-aristizabal) [inline src/gval/comparison/compute_continuous_metrics.py:23]: I know this function existed but based on my understanding would something like `_get_data_mask` be a more appropriate name for it? The same could be said about the top level description in the docstring.

File: src/gval/utils/loading_datasets.py [hunk 4]
```diff
@@ -229,6 +238,200 @@ def _convert_to_dataset(xr_object=Union[xr.DataArray, xr.Dataset]) -> xr.Dataset
         return xr_object
 
 
+def _get_raster_band_nodata(band_metadata, nodata_fill) -> Number:
+    """
```
Comment (fernando-aristizabal) [inline src/gval/utils/loading_datasets.py:242]: Top level description?

File: src/gval/utils/loading_datasets.py [hunk 5]
```diff
@@ -229,6 +238,200 @@ def _convert_to_dataset(xr_object=Union[xr.DataArray, xr.Dataset]) -> xr.Dataset
...
+
+    Parameters
+    ----------
+    band_metadata : list
+        Metadata from raster:bands extension
+    nodata_fill : Number
+        Fill in value for missing data
```
Comment (fernando-aristizabal) [inline src/gval/utils/loading_datasets.py:249]: Raises section?

File: src/gval/utils/loading_datasets.py [hunk 6]
```diff
@@ -229,6 +238,200 @@ def _convert_to_dataset(xr_object=Union[xr.DataArray, xr.Dataset]) -> xr.Dataset
...
+            )
+
+        return nodata_fill
+
+
+def _set_nodata(
+    stack: xr.DataArray, band_metadata: list = None, nodata_fill: Number = None
+) -> Number:
+    """
+
```
Comment (fernando-aristizabal) [inline src/gval/utils/loading_datasets.py:275]: Some top level descriptions would be good for all these functions.

File: src/gval/utils/loading_datasets.py [hunk 7]
```diff
@@ -229,6 +238,200 @@ def _convert_to_dataset(xr_object=Union[xr.DataArray, xr.Dataset]) -> xr.Dataset
...
+    """
+
+    if stack.rio.crs is not None:
+        return stack.rio.write_crs(stack.rio.crs)
+    else:
+        return stack.rio.write_crs(f"EPSG:{band_metadata['epsg'].values}")
+
+
+def get_stac_data(
```
Comment (fernando-aristizabal) [inline src/gval/utils/loading_datasets.py:319]: Should we be returning stac's as catalogs or xarray's? I believed the original thought was to return catalogs in the panda's DF style?

Or is this the lower level function and some future function will turn filtered STAC objects into catalog DF's?

File: notebooks/Catalog Tutorial.ipynb
Comment (fernando-aristizabal) [inline notebooks/Catalog Tutorial.ipynb:1]: Possibly consider introducing an attribute column to both candidate and benchmark catalogs to show how this works?

File: notebooks/Subsampling Tutorial.ipynb
Comment (fernando-aristizabal) [inline notebooks/Subsampling Tutorial.ipynb:1]: Looks nice to see this in action!

**Review by fernando-aristizabal — CHANGES_REQUESTED** [review]

File: src/gval/utils/loading_datasets.py (hunk 7 above)
Comment (GregoryPetrochenkov-NOAA) [reply src/gval/utils/loading_datasets.py:319]: This particular function was made to get an online asset directly in to an xarray object.  In the event that a catalog with STAC items was created as you are referring to, then at process time this function would be able to create xarray objects for each asset in the DataFrame and then do the actual comparisons.

File: README.MD (hunk 2 above)
Comment (GregoryPetrochenkov-NOAA) [reply README.MD:76]: Changed wording as requested.

File: src/gval/comparison/compute_continuous_metrics.py (hunk 3 above)
Comment (GregoryPetrochenkov-NOAA) [reply src/gval/comparison/compute_continuous_metrics.py:23]: I went ahead and changed the name of the function and docstring.

File: src/gval/utils/loading_datasets.py (hunk 4 above)
Comment (GregoryPetrochenkov-NOAA) [reply src/gval/utils/loading_datasets.py:242]: Added top level description

File: src/gval/utils/loading_datasets.py (hunk 5 above)
Comment (GregoryPetrochenkov-NOAA) [reply src/gval/utils/loading_datasets.py:249]: Added Raises Section

File: README.MD (hunk 1 above)
Comment (GregoryPetrochenkov-NOAA) [reply README.MD:165]: I went ahead and added catalog level attributes to both candidate and benchmark catalogs.

File: docs/images/benchmark_catalog.png
Comment (GregoryPetrochenkov-NOAA) [reply docs/images/benchmark_catalog.png:1]: I went ahead and added catalog attributes to both candidate and benchmark catalogs.

File: docs/images/candidate_catalog.png
Comment (GregoryPetrochenkov-NOAA) [reply docs/images/candidate_catalog.png:1]: I went ahead and added catalog attributes to both candidate and benchmark catalogs.

File: notebooks/Catalog Tutorial.ipynb
Comment (GregoryPetrochenkov-NOAA) [reply notebooks/Catalog Tutorial.ipynb:1]: I went ahead and added catalog attributes to both candidate and benchmark catalogs.

**Review by fernando-aristizabal — APPROVED** [review]

**Pull request merged**